### PR TITLE
[UAT] scale text on the variable browser plot with the plot hieght and width #312

### PR DESCRIPTION
**Pull request by chlebowa**
Closes [this issue](https://github.com/insightsengineering/teal.modules.general/issues/312).

Added collapsible slider to allow user to manipulate text size in plots.
Patched applying theme to plots.
Added aliases for module documentation.

### CODE REVIEW

File: R/tm_variable_browser.R [hunk 1]
```diff
@@ -156,7 +158,18 @@ ui_variable_browser <- function(id,
               } else {
                 shinyjs::hidden(x)
               }
-            }
+            },
+            # input user-defined text size
+            teal.widgets::panel_item(
+              title = "Plot settings",
+              collapsed = TRUE,
+              teal.widgets::optionalSliderInputValMinMax(
```
Comment (nikolas-burkoff) [inline R/tm_variable_browser.R:166]: This is fine - I might have just used a plain shiny sliderInput

**Review by nikolas-burkoff — APPROVED** [review]
Looking good:

A few comments:
1) Please add a NEWS item
2) I wonder if we should move the ui to here?
image - I'll leave that up to you
3) I wonder how much effort it is to add the "theme" option like we do in the other tmg modules - probably not worth adding tbh unless it's super easy

File: R/tm_variable_browser.R [hunk 2]
```diff
@@ -104,6 +112,7 @@ ui_variable_browser <- function(id,
 
   shiny::tagList(
     include_css_files("custom"),
+    shinyjs::useShinyjs(),
```
Comment (nikolas-burkoff) [inline R/tm_variable_browser.R:115]: So no harm putting this here but it is called [here](https://github.com/insightsengineering/teal/blob/eb373dfdfb06aaabc97a10b7c5c7ce082d6429e3/R/include_css_js.R#L80) from [here](https://github.com/insightsengineering/teal/blob/79678e78267098d1a0377f0b6b9cd8bba1cea988/R/module_teal.R#L95)

**Comment by chlebowa** [comment]
> I wonder how much effort it is to add the "theme" option like we do in the other tmg modules - probably not worth adding tbh unless it's super easy

That would require rendering the UI for plot settings server side as a default theme may be supplied by `ggplot2_args`. Not too hard, but not super easy either.

**Review by nikolas-burkoff — APPROVED** [review]

File: R/tm_variable_browser.R (hunk 2 above)
Comment (chlebowa) [reply R/tm_variable_browser.R:115]: Ah, I was wondering whether `shinyjs` is added to all apps by default.

**Comment by chlebowa** [comment]
>> I wonder how much effort it is to add the "theme" option like we do in the other tmg modules - probably not worth adding tbh unless it's super easy

> That would require rendering the UI for plot settings server side as a default theme may be supplied by ggplot2_args. Not too hard, but not super easy either.

Actually, no it wouldn't, as a theme is never passed as a complete theme, but a set of element definitions in `ggplot2_args`. It's easy enough.

But I ran into another problem there, an assertion fails that probably shouldn't. Consulting with @Polkas .

**Comment by chlebowa** [comment]
Adding theme selection had two unexpected effects:
1) revealed problem with argument checks in `resolve_ggplot2_args`, described in detail [here](https://github.com/insightsengineering/teal.widgets/issues/108)
2) overwrote the dev-specified x label rotation of 45 degrees; I added another input for that to the UI so that user settings take precedence before theme

File: R/tm_variable_browser.R [hunk 3]
```diff
@@ -246,6 +264,24 @@ srv_variable_browser <- function(id,
...
+    # add used-defined text size to ggplot arguments passed from caller frame
+    all_ggplot2_args <- reactive({
+      user_text <- teal.widgets::ggplot2_args(
+        theme = list("text" = ggplot2::element_text(size = input[["font_size"]]),
+                     "axis.text.x" = ggplot2::element_text(angle = input[["label_rotation"]], hjust = 1))
+      )
+      user_theme <- getFromNamespace(sprintf("theme_%s", input[["ggplot_theme"]]), ns = "ggplot2")
+      user_theme <- user_theme()
+      # temporary fix to circumvent assertion issue with resolve_ggplot2_args
+      # drop problematic elements
```
Comment (nikolas-burkoff) [inline R/tm_variable_browser.R:276]: https://github.com/insightsengineering/teal.widgets/issues/108

File: R/tm_variable_browser.R [hunk 4]
```diff
@@ -175,6 +173,21 @@ ui_variable_browser <- function(id,
...
+              title = "Plot settings",
+              collapsed = TRUE,
+              selectInput(inputId = ns("ggplot_theme"), label = "ggplot2 theme",
+                          choices = c("gray", "bw", "linedraw", "light",
+                                      "dark", "minimal", "classic", "void", "test"),
+                          selected = "grey"),
+              fluidRow(
+                column(6, sliderInput(inputId = ns("font_size"), label = "font size",
+                                      min = 5L, max = 30L, value = 15L, step = 1L, ticks = FALSE)),
+                column(6, sliderInput(inputId = ns("label_rotation"), label = "rorate x labels",
```
Comment (nikolas-burkoff) [inline R/tm_variable_browser.R:187]:
```suggestion
                column(6, sliderInput(inputId = ns("label_rotation"), label = "rotate x labels",
```

**Pull request merged**